sdk/storage/Azure.Storage.Blobs/src/PageBlobWriteStream.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Azure.Storage.Shared;
+
+namespace Azure.Storage.Blobs
+{
+    internal class PageBlobWriteStream : WriteStream
+    {
+        private readonly PageBlobClient _pageBlobClient;
+        private readonly PageBlobRequestConditions _conditions;
+        private long _writeIndex;
+
+        public PageBlobWriteStream(
+            PageBlobClient pageBlobClient,
+            long bufferSize,
+            long position,
```

Comment:
https://github.com/Azure/azure-sdk-for-net/pull/13344/files?file-filters%5B%5D=.cs#diff-e39f8ee2730163a6c9c2d682c1e9c025R127

Reply:
Position, not buffer size. I don't see position getting passed to any validation method in this or the base constructor.

Regarding `ValidateBufferSize`, it's abstract. Why don't we call the method in the base constructor instead of repeatedly in each subclass?

Reply:
Added `ValidatePosition()` to `PageBlobWriteStream` in the next iteration.

I don't call `ValidateBufferSize()` in the base constructor because:

`CA2214: Do not call overridable methods in constructors `.